### PR TITLE
[nvidia-bluefield] Add support for new "nvidia-bluefield" platform.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: platform/nvidia-bluefield/recipes/gh-helpers.mk [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+BF_ASSETS_GITHUB_REPO = nvidia-sonic/sonic-bluefield-packages
+SDK_ASSETS_GITHUB_TOKEN =
```
Comment (vivekrnv) [inline platform/nvidia-bluefield/recipes/gh-helpers.mk:19]: How do we plan to handle this, shouldn't we be hosting these packages in a public repo instead?

Comment (oleksandrivantsiv) [reply platform/nvidia-bluefield/recipes/gh-helpers.mk:19]: Yes, I'm working on moving releases to the public repository.

**Review by vivekrnv — APPROVED** [review]

File: platform/nvidia-bluefield/platform-api/sonic_platform/device_data.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    def get_sfp_data(self, sfp_index):
+        if sfp_index >= len(self._chassis['sfps']):
+            raise RuntimeError("Wrong SFP index: {}".format(sfp_index))
+        sfp_data = self._chassis['sfps'][sfp_index]
+        return SfpData(**sfp_data)
+
+    def get_bf_version(self):
+        bf2_platforms = ['Nvidia-MBF2H536C']
+        return BFVersion.BF2 if self.get_platform_name() in bf2_platforms else BFVersion.BF3
```
Comment (Yakiv-Huryk) [inline platform/nvidia-bluefield/platform-api/sonic_platform/device_data.py]: do we need it?

File: platform/nvidia-bluefield/platform-api/sonic_platform/thermal.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#############################################################################
+
+try:
+    from .device_data import BFVersion
+    from .thermal_bf3 import initialize_chassis_thermals as bf3_init
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+def initialize_chassis_thermals(bfversion):
+    if bfversion == BFVersion.BF3:
```
Comment (Yakiv-Huryk) [inline platform/nvidia-bluefield/platform-api/sonic_platform/thermal.py]: if we don't support BF2, maybe it's better to remove BFVersion and these checks altogether?

File: platform/nvidia-bluefield/platform-api/sonic_platform/device_data.py (hunk 2 above)
Comment (oleksandrivantsiv) [reply platform/nvidia-bluefield/platform-api/sonic_platform/device_data.py]: no, removed

File: platform/nvidia-bluefield/platform-api/sonic_platform/thermal.py (hunk 3 above)
Comment (oleksandrivantsiv) [reply platform/nvidia-bluefield/platform-api/sonic_platform/thermal.py]: removed

**Pull request closed without merging**